.github/workflows/main.yaml

```diff
@@ -53,3 +60,25 @@ jobs:
           fail_ci_if_error: true
           files: ./coverage.xml
           flags: unittests
+  build-and-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+    runs-on: ubuntu-latest
+    needs:
+      - unit-tests
+      - integration-tests
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
+        with:
```

Comment:
you should be able to add cache:true. Then it doesn't redownload all dependencies with every run